xlators/storage/posix/src/posix-inode-fd-ops.c

```diff
@@ -5488,8 +5488,8 @@ posix_fentrylk(call_frame_t *frame, xlator_t *this, const char *volume,
     return 0;
 }
 
-int
-posix_fill_readdir(fd_t *fd, DIR *dir, off_t off, size_t size,
+static int
+posix_fill_readdir(fd_t *fd, struct posix_fd *pfd, off_t off, size_t size,
```

Comment:
I think now we don't need to pass fd_t in posix_fill_readdir, earlier we do pass it as a argument because we do access the ctx from fd_object but now are directly passing the ctx as a function argument so it is not required.